pkg/netpol/ingressanalyzer/ingress_analyzer.go

```diff
@@ -0,0 +1,200 @@
+		case scan.Route:
+			err = ia.mapRouteToServices(obj.Route)
+		}
+		if err != nil {
+			return nil, err
+		}
+	}
+	return ia, err
+}
+
+/////////////////////////////////////////////////////////////////////////////////////////////////
+// services analysis
+
+const (
+	selectorError        = "selector conversion error"
+	missingSelectorError = "error : K8s Service without selectors is not supported"
+)
+
+// this function populates servicesToPeersMap
+func (ia *IngressAnalyzer) mapServiceToPeers(svc *corev1.Service) error {
+	// get peers selected by the service selectors
+	peers, err := ia.getServicePeers(svc)
+	if err != nil {
+		return err
+	}
+	if _, ok := ia.servicesToPeersMap[svc.Namespace]; !ok {
+		ia.servicesToPeersMap[svc.Namespace] = make(map[string][]eval.Peer)
+	}
+	ia.servicesToPeersMap[svc.Namespace][svc.Name] = peers
+	return nil
+}
+
+func (ia *IngressAnalyzer) getServicePeers(svc *corev1.Service) ([]eval.Peer, error) {
+	svcStr := types.NamespacedName{Name: svc.Name, Namespace: svc.Namespace}.String()
+	if svc.Spec.Selector == nil {
+		return nil, errors.New(scan.Service + " " + svcStr + " " + missingSelectorError)
+	}
+	svcLabelsSelector, err := convertServiceSelctorToLabelSelector(svc.Spec.Selector, svcStr)
+	if err != nil {
+		return nil, err
+	}
+	return ia.pe.GetSelectedPeers(svcLabelsSelector, svc.Namespace)
+}
+
+// utility func
+func convertServiceSelctorToLabelSelector(svcSelect map[string]string, svcStr string) (labels.Selector, error) {
+	labelsSelector := metav1.LabelSelector{MatchLabels: svcSelect}
+	selectorRes, err := metav1.LabelSelectorAsSelector(&labelsSelector)
+	if err != nil {
+		return nil, errors.New(scan.Service + " " + svcStr + " " + selectorError)
+	}
+	return selectorRes, nil
+}
+
+// //////////////////////////////////////////////////////////////////////////////////////////////
+// routes analysis
+
+const (
+	maxBackendServices = 3
+	allowedTargetKind  = scan.Service
+	routeTargetKindErr = "target kind error"
+	routeBackendsErr   = "alternate backends error"
+)
+
+// this function populates routesToServicesMap
+func (ia *IngressAnalyzer) mapRouteToServices(rt *ocroutev1.Route) error {
+	services, err := getRouteServices(rt)
+	if err != nil {
+		return err
+	}
+	if _, ok := ia.routesToServicesMap[rt.Namespace]; !ok {
+		ia.routesToServicesMap[rt.Namespace] = make(map[string][]string)
+	}
+	ia.routesToServicesMap[rt.Namespace][rt.Name] = services
+	return nil
+}
+
+func getRouteServices(rt *ocroutev1.Route) ([]string, error) {
+	routeStr := types.NamespacedName{Namespace: rt.Namespace, Name: rt.Name}.String()
+	// Currently, only 'Service' is allowed as the kind of target that the route is referring to.
+	if rt.Spec.To.Kind != "" && rt.Spec.To.Kind != allowedTargetKind {
+		return nil, errors.New(scan.Route + " " + routeStr + ": " + routeTargetKindErr)
+	}
+	if len(rt.Spec.AlternateBackends) > maxBackendServices {
+		return nil, errors.New(scan.Route + " " + routeStr + ": " + routeBackendsErr)
+	}
```

Comment:
why do we need to check this?

Reply:
according to [this](https://pkg.go.dev/github.com/openshift/api/route/v1#RouteSpec) - 

> // alternateBackends allows up to 3 additional backends to be assigned to the route.

I assumed we need to check it

Reply:
can remove this check

Reply:
done